lib/web_ui/test/canvaskit/initialization/load_manifest_fonts_manually_test.dart

```diff
@@ -0,0 +1,94 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:js_interop';
+import 'dart:typed_data';
+
+import 'package:test/bootstrap/browser.dart';
+import 'package:test/test.dart';
+import 'package:ui/src/engine.dart';
+import 'package:ui/ui.dart' as ui;
+import 'package:ui/ui_web/src/ui_web.dart' as ui_web;
+import 'package:web_engine_tester/golden_tester.dart';
+
+@JS('window.flutterConfiguration')
+external set _jsConfiguration(JsFlutterConfiguration? configuration);
+
+void main() {
+  internalBootstrapBrowserTest(() => testMain);
+}
+
+void testMain() {
+  test('fonts can be loaded manually', () async {
+    _jsConfiguration = null;
+
+    // Fully initialize engine services and UI without manifest fonts. Test that
+    // the engine can still render everything, albeit without text.
+    await initializeEngineServices(jsConfiguration: JsFlutterConfiguration(
+      loadManifestFontsBeforeAppMain: false.toJS,
+      canvasKitBaseUrl: '/canvaskit/'.toJS,
+    ));
+    await initializeEngineUi();
+
+    String? messageChannel;
+    Object? messageData;
+
+    EnginePlatformDispatcher.instance.onPlatformMessage = (String name, ByteData? data, ui.PlatformMessageResponseCallback? callback) {
+      messageChannel = name;
+      messageData = const JSONMessageCodec().decodeMessage(data);
+    };
+
+    await _testTextRendering('canvaskit_text_without_fonts');
+    expect(messageChannel, isNull);
+    expect(messageData, isNull);
+
+    // Load fonts manually. Test that text is now rendered fully.
+    await ui_web.loadManifestFonts();
+
+    // The notification message is sent in the next frame so there's only one
+    // notification for a batch of fonts, so wait for 2 frames before checking
+    // for the message.
+    final Completer<void> waitForFontMessage = Completer<void>();
+    domWindow.requestAnimationFrame((_) {
+      domWindow.requestAnimationFrame((_) {
+        waitForFontMessage.complete();
+      });
+    });
```

Comment:
Why not make the `onPlatformMessage` a function that funnels the data to a Stream, and here you just `await` for that?

Relying on X numbers of frames seems flaky. For example: Why 2 if we've already awaited for the fonts to be loaded above?